pageserver/src/tenant/metadata.rs

```diff
@@ -14,10 +14,19 @@ use utils::bin_ser::SerializeError;
 use utils::{bin_ser::BeSer, id::TimelineId, lsn::Lsn};
 
 /// Use special format number to enable backward compatibility.
-const METADATA_FORMAT_VERSION: u16 = 4;
+///
+/// Currently, we are in the transition period between bodyv3 and bodyv2. We will keep
+/// both formats available for a week, and then auto-upgrade everything to v3.
+///
+/// The version is distinguished by the header format number. We always use bodyv3 as
+/// the in-memory representation of the data. If the header format is the old one, the
+/// body will be serialized using bincode. Otherwise, it will be serialized using JSON,
+/// which is the default method for bodyv3.
+const METADATA_FORMAT_VERSION: u16 = 5;
```

Comment:
Should we make an enum for these format versions?

Reply:
(Don't have to do it in this PR, just general question)

Reply:
yep, I agree -- with more versions in the future, there should be an enum to map from FORMAT_VERSION_VX to a number.